x-pack/plugins/security_solution/public/management/pages/policy/view/policy_advanced.tsx

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useCallback, useState, useMemo, useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import {
+  EuiFieldText,
+  EuiText,
+  EuiIconTip,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { cloneDeep } from 'lodash';
+import { policyConfig } from '../store/policy_details/selectors';
+import { usePolicyDetailsSelector } from './policy_hooks';
+
+function setValue(obj: Record<string, unknown>, value: string, path: string[]) {
```

Comment:
in order to not send advanced values to the Endpoint unless the user specifically sets them, we could something like this.  (Code can be cleaned up.)

```js
function setValue(obj: Record<string, unknown>, value: string, path: string[]) {
  let newPolicyConfig = obj;
  for (let i = 0; i < path.length - 1; i++) {
    if(!newPolicyConfig[path[i]]) {
      newPolicyConfig[path[i]] = {} as Record<string, unknown>;
    }
    newPolicyConfig = newPolicyConfig[path[i]] as Record<string, unknown>;
  }
  newPolicyConfig[path[path.length - 1]] = value;
}
```

Reply:
23238e9d29504f34a9e774cad99443b830951c52